coins/monero/tests/add_data.rs

```diff
@@ -0,0 +1,80 @@
+use monero_serai::{rpc::Rpc, wallet::TransactionError, transaction::Transaction};
+
+mod runner;
+
+test!(
+  add_single_data_less_than_255,
+  (
+    |_, mut builder: Builder, addr| async move {
+      // make a data that is less than 255 bytes
+      let arbitrary_data = Vec::from("this is an arbitrary data less than 255 bytes");
+
+      // make sure we can add to tx
+      let result = builder.add_data(&arbitrary_data);
+      assert!(result.is_ok());
+
+      builder.add_payment(addr, 5);
+      (builder.build().unwrap(), (arbitrary_data,))
```

Comment:
The state doesn't need to be a tuple.

Reply:
yes. Just a preference. So that it structure is the same if I need to pass multiple variables!